### PR TITLE
Fixes in OOI Add/ Edit form

### LINKED ISSUE #2901: Error when manually adding a OOIs
**Describe the bug**
Various OOIs cannot be added manually as they trigger errors. Screenshots and stacktrace outputs are shown below as well. The following OOIs were identified to be broken:

**- Question object a TypeError is returned.** 
```
Exception Value: 	

issubclass() arg 1 must be a class

Exception Location: 	/app/rocky/tools/forms/ooi_form.py, line 73, in generate_form_fields
```

**- AutonomousSystem number**  
```
Exception Value: 	

'OOIForm' has no field named 'name'.

Exception Location: 	/usr/local/lib/python3.11/site-packages/django/forms/forms.py, line 400, in add_error
```

**- Config -   `'OOIForm' has no field named 'config'.`**

```
Exception Value: 	

'OOIForm' has no field named 'config'.

Exception Location: 	/usr/local/lib/python3.11/site-packages/django/forms/forms.py, line 400, in add_error
```

- **Incident 'OOIForm' has no field named 'meta_data'.**
```
Exception Value: 	

'OOIForm' has no field named 'meta_data'.

Exception Location: 	/usr/local/lib/python3.11/site-packages/django/forms/forms.py, line 400, in add_error
```

**To Reproduce**
1. Go to Object.
2. Add object of type 'Question'
3. See error

**Expected behavior**
The chosen object can be added. 

**Screenshots**

image

**OpenKAT version**
commit 3588c77f085b5643823477b1636ed33a37d92302 (HEAD -> main, origin/main, origin/HEAD)

**Stacktrace**
```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/en/aa/objects/add/Question/

Django Version: 4.2.11
Python Version: 3.11.9
Installed Applications:
['whitenoise.runserver_nostatic',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.forms',
 'django_otp',
 'django_otp.plugins.otp_static',
 'django_otp.plugins.otp_totp',
 'two_factor',
 'account',
 'tools',
 'fmea',
 'crisis_room',
 'onboarding',
 'katalogus',
 'django_password_validators',
 'django_password_validators.password_history',
 'rest_framework',
 'tagulous',
 'compressor',
 'reports',
 'knox',
 'csp']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django_otp.middleware.OTPMiddleware',
 'rocky.middleware.auth_required.AuthRequiredMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'rocky.middleware.onboarding.OnboardingMiddleware',
 'csp.middleware.CSPMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 144, in get
    return self.render_to_response(self.get_context_data())
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/rocky/views/ooi_add.py", line 69, in get_context_data
    context = super().get_context_data(**kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/account/mixins.py", line 103, in get_context_data
    context = super().get_context_data(**kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 74, in get_context_data
    kwargs["form"] = self.get_form()
                     ^^^^^^^^^^^^^^^
  File "/app/rocky/rocky/views/ooi_view.py", line 147, in get_form
    form = super().get_form(form_class)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 39, in get_form
    return form_class(**self.get_form_kwargs())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/tools/forms/ooi_form.py", line 28, in __init__
    fields = self.get_fields()
             ^^^^^^^^^^^^^^^^^
  File "/app/rocky/tools/forms/ooi_form.py", line 36, in get_fields
    return self.generate_form_fields()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/rocky/tools/forms/ooi_form.py", line 73, in generate_form_fields
    elif self.ooi_class == Question and issubclass(annotation, str) and name == "json_schema":
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^

Exception Type: TypeError at /en/aa/objects/add/Question/
Exception Value: issubclass() arg 1 must be a class
```

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
This PR fixes some rendering and validation issues in the OOI add/ edit form.

### Issue link
Closes #2901 

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
looks good to me.

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Manually added the Question, ASnumber and Config OOIs and they are created without an error. I have not verified if the output from these OOIs is parsed correctly, as our documentation is outdated on this part. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

File: octopoes/octopoes/models/ooi/monitoring.py [hunk 1]
```diff
@@ -19,6 +21,6 @@ class Incident(OOI):
     event_type: str
     event_title: str
     severity: str
-    meta_data: dict[str, Any]
+    meta_data: JsonValue
```
Comment (dekkers) [inline octopoes/octopoes/models/ooi/monitoring.py:24]: Shouldn't this be the following like the change in config.py?

```suggestion
    meta_data: dict[str, JsonValue]
```

Comment (ammar92) [reply octopoes/octopoes/models/ooi/monitoring.py:24]: I wasn't sure whether this field allows full JSON values or just a key-value/ object. I can change it to `dict[str, JsonValue]`?

Comment (ammar92) [reply octopoes/octopoes/models/ooi/monitoring.py]: Implemented the suggestion, it seems indeed as if `Incident.meta_data` is filled with key-value pairs.

**Review by dekkers — APPROVED** [review]

**Pull request merged**